getting-started/release-notes/changes-in-timescaledb-2.md

```diff
@@ -0,0 +1,290 @@
+
+## Using data retention and continuous aggregates together [](retention-and-caggs)
+
+With greater power and flexibility also comes a greater responsibility to understand how features interact. Specifically, users should understand the interactions between data retention policy settings and continuous aggregate settings. 
+
+Before starting the upgrade to TimescaleDB 2.0, **we highly recommend checking the database log for errors related to failed retention policies that were occurring in TimescaleDB 1.x** and then either remove them or update them to be compatible with existing continuous aggregates. Any remaining retention policies that are still incompatible with the `ignore_invalidation_older_than `setting will automatically be disabled with a notice during the upgrade.
+
+As an example, if a data retention policy on a hypertable is set for `drop_after => '4 weeks'`, then the policy associated with a continuous aggregate on that same hypertable should have a `start_offset` less than or equal to 4 weeks.  Similarly, any manual call to `refresh_continuous_aggregate` should likely specify a `window_start` that’s also less than the date from 4 weeks ago. 
+
+If not understood properly, users could overwrite existing aggregated data in continuous aggregates with empty data by recomputing data over a now-dropped time window of data, not likely the result a user was expecting. Instead, users seeking to keep only a certain time window of data in their continuous aggregate view should create a data retention policy on the continuous aggregate.
+
+**This differs significantly from how TimescaleDB 1.x handled conflicts between retention policies and continuous aggregates.**
+
+Previously, if the continuous aggregation setting `ignore_invalidation_older_than` overlapped with data that would be dropped by a retention policy, the retention policy would silently fail.  Making the retention policy work again required users to modify settings in either the retention policy or the continuous aggregate, and even then some data wasn’t always materialized as expected. 
+
+After upgrading to TimescaleDB 2.0, **retention policies will no longer fail due to incompatibilities with continuous aggregates** and users have to ensure that retention and continuous aggregate policies have the desired interplay. 
+
+## Differences in the handling of backfill on continuous aggregates
+
+In TimescaleDB 1.x, data that was backfilled into hypertables wasn’t handled optimally; modification of any hypertable data, regardless of how old, would cause the continuous aggregate materializer job to restart from the point of the earliest backfill and then it would materialize from that point forward. Unfortunately, this could also cause the materializer to get “stuck”, since the background job only processed a limited amount of data per run, as governed by the `max_interval_per_job` setting. When this happened, one job run could potentially force the next job to start all over again. 
```

Comment:
```suggestion
In TimescaleDB 1.x, data that was backfilled into hypertables wasn’t handled optimally; modification of any hypertable data, regardless of how old, would cause the continuous aggregate materializer job to restart from the point of the earliest backfill and then materialize from that point forward. Unfortunately, this could also cause the materializer to get “stuck”, since the background job only processed a limited amount of data per run, as governed by the `max_interval_per_job` setting. When this happened, one job run could potentially force the next job to start all over again. 
```